core/scripts/collateral-umip/1_Propose.js

```diff
@@ -0,0 +1,73 @@
+// This script generates and submits an identifier-add upgrade transaction to the DVM. It can be run on a local ganache
+// fork of the main net or can be run directly on the main net to execute the upgrade transactions.
+// To run this on the localhost first fork main net into Ganache with the proposerWallet unlocked as follows:
+// ganache-cli --fork https://mainnet.infura.io/v3/d70106f59aef456c9e5bfbb0c2cc7164 --unlock 0x2bAaA41d155ad8a4126184950B31F50A1513cE25
+// Then execute the script as: truffle exec ./scripts/collateral-umip/1_Propose.js --network mainnet-fork --collateral 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 --fee 0.1 from core
+
+const AddressWhitelist = artifacts.require("AddressWhitelist");
+const Store = artifacts.require("Store");
+const Governor = artifacts.require("Governor");
+
+const argv = require("minimist")(process.argv.slice(), { string: ["collateral", "fee"] });
+
+const proposerWallet = "0x2bAaA41d155ad8a4126184950B31F50A1513cE25";
+
+async function runExport() {
+  console.log("Running Upgrade🔥");
+  console.log("Connected to network id", await web3.eth.net.getId());
+
+  if (!argv.collateral || !argv.fee) {
+    throw "Must provide --fee and --collateral";
+  }
+
+  // The proposal will first add a final fee for the currency.
+  const store = await Store.deployed();
+  const addFinalFeeToStoreTx = store.contract.methods
```

Comment:
OOC: Is there a way to check that these transactions would succeed? So that we don't pass a vote only for the transaction to fail. You could call `.send()` instead of `.encodeABI()` and just check the output is a success state?

Reply:
Using the three scripts, I simulated them against a mainnet fork manually.

IMO, we should probably add an item to the backlog to begin simulating these in ci (using a separate fork-based job) to automate these tests.

Thoughts?

Reply:
how would this look in CI? you would want to re-run these script to ensure that they always produce the desired state changes? is this useful given the scripts really are only used _once_ when making the initial add? I guess that it would ensure that the scripts continue to operate as expected.

Reply:
Yeah, this could make sense for the scripts that are re-usable, but I think what you're getting at is that the one-off upgrades don't need to be repeatedly tested, right?